src/axom/inlet/tests/inlet_object.cpp

```diff
@@ -939,6 +939,69 @@ TYPED_TEST(inlet_object, struct_arrays_as_std_vector)
   EXPECT_EQ(foos, expected_foos);
 }
 
+TYPED_TEST(inlet_object, default_scalar_marked_false)
+{
+  std::string testString = " ";
+  DataStore ds;
+  Inlet inlet = createBasicInlet<TypeParam>(&ds, testString);
+
+  auto& scalar = inlet.addInt("foo").defaultValue(2);
+  // The field itself should evaluate to false because nothing was provided
+  auto& field = static_cast<axom::inlet::Field&>(scalar);
+  EXPECT_FALSE(static_cast<bool>(field));
```

Comment:
The semantics of casting `bool()` to `false` when using a default value seems odd to me, 
e.g. as opposed to a function like `Field::isUserProvided()`  (but obviously a better name than that).